src/librustc_typeck/outlives/mod.rs

```diff
@@ -46,14 +42,43 @@ fn inferred_outlives_of<'a, 'tcx>(
         hir_map::NodeItem(item) => match item.node {
             hir::ItemStruct(..) | hir::ItemEnum(..) | hir::ItemUnion(..) => {
                 let crate_map = tcx.inferred_outlives_crate(LOCAL_CRATE);
-                let dep_node = item_def_id.to_dep_node(tcx, DepKind::InferredOutlivesOf);
-                tcx.dep_graph.read(dep_node);
+                // let dep_node = item_def_id.to_dep_node(tcx, DepKind::InferredOutlivesOf);
+                // tcx.dep_graph.read(dep_node);
 
-                crate_map
+                let predicates = crate_map
```

Comment:
Ah, I think the order problem comes from the fact that this is a hashmap here, and hence iteration order is undefined. That's.. probably good news, because I think its effects should be limited? In any case, for the *test output*, I would sort the vec of strings (see comment below).